File: backend/database/inspect_demo.py
import sqlite3
import os

db_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'demoTable.db')
print(f'Opening: {db_path}')
print(f'Exists: {os.path.exists(db_path)}')

conn = sqlite3.connect(db_path)
cur = conn.cursor()

# List tables
cur.execute("SELECT name, sql FROM sqlite_master WHERE type='table'")
tables = cur.fetchall()

print('\n=== TABLES ===')
for name, sql in tables:
    cur.execute(f'SELECT COUNT(*) FROM [{name}]')
    count = cur.fetchone()[0]
    print(f'\nTABLE: {name} ({count} rows)')
    print(sql)

# Column info
print('\n=== COLUMNS ===')
for name, _ in tables:
    cur.execute(f'PRAGMA table_info([{name}])')
    cols = cur.fetchall()
    print(f'\n{name}:')
    for col in cols:
        print(f'  {col[1]:20s} {col[2]:15s} {"NOT NULL" if col[3] else "":10s} {"PK" if col[5] else ""}')

# Sample rows
print('\n=== SAMPLE DATA (first 3 rows per table) ===')
for name, _ in tables:
    cur.execute(f'SELECT * FROM [{name}] LIMIT 3')
    rows = cur.fetchall()
    print(f'\n{name}:')
    # Get column names
    col_names = [desc[0] for desc in cur.description]
    print(f'  Columns: {col_names}')
    for row in rows:
        print(f'  {row}')

# Last 3 rows too
print('\n=== LAST 3 ROWS (to see time range) ===')
for name, _ in tables:
    cur.execute(f'SELECT * FROM [{name}] ORDER BY rowid DESC LIMIT 3')
    rows = cur.fetchall()
    print(f'\n{name}:')
    for row in rows:
        print(f'  {row}')

conn.close()
print('\nDone.')

File: backend/database/inspect_demo.js
// Temporary script to inspect demoTable.db schema
const path = require('path');
const Database = require(path.join(__dirname, '..', 'node_modules', 'better-sqlite3'));

const dbPath = path.join(__dirname, 'demoTable.db');
console.log('Opening:', dbPath);

const db = new Database(dbPath);

// List all tables
const tables = db.prepare("SELECT name, sql FROM sqlite_master WHERE type='table'").all();
console.log('=== TABLES ===');
tables.forEach(t => {
    console.log('\nTABLE:', t.name);
    console.log(t.sql);
    const rows = db.prepare('SELECT COUNT(*) as cnt FROM [' + t.name + ']').get();
    console.log('Rows:', rows.cnt);
});

// Show sample rows from each table
console.log('\n=== SAMPLE DATA ===');
tables.forEach(t => {
    console.log('\nTABLE:', t.name, '- First 3 rows:');
    const rows = db.prepare('SELECT * FROM [' + t.name + '] LIMIT 3').all();
    rows.forEach(r => console.log(JSON.stringify(r)));
});

// Also show column info
console.log('\n=== COLUMN INFO ===');
tables.forEach(t => {
    console.log('\nTABLE:', t.name);
    const info = db.prepare('PRAGMA table_info([' + t.name + '])').all();
    info.forEach(c => console.log('  ', c.name, c.type, c.notnull ? 'NOT NULL' : '', c.pk ? 'PK' : ''));
});

db.close();
console.log('\nDone.');
